### ISSUE TITLE
Implement drone node partitioning

### ISSUE THREAD

**Issue by nwconstable** [nwconstable/sdsr-site#12]
**File:** `project/src/partition.py`
**Depends on:** build_graph.py (done)

### Context
The grid graph produced by `build_pyg_data` must be split into K spatial
regions, one per simulated drone. Each drone trains only on its local subgraph.
Partitioning is row-wise strip splitting (column-major strips along the x-axis).

**Note — existing implementation:** A private `_build_local_subgraph` helper
already exists in `train.py` (line ~71):
```python
def _build_local_subgraph(data: Data, node_indices: np.ndarray) -> Data:
    global_idx = torch.tensor(node_indices, dtype=torch.long)
    sub_ei, _ = pyg_subgraph(global_idx, data.edge_index,
                              relabel_nodes=True, num_nodes=data.num_nodes)
    return Data(x=data.x[global_idx], edge_index=sub_ei, y=data.y[global_idx])
```
`partition.py` must provide the **canonical public** version of this function.
Once `partition.py` is implemented, update `train.py` to `from partition import
build_local_subgraph` and remove the private duplicate. `federated_agent.py`
uses an inline equivalent — update it to import from `partition.py` as well.

### Required functions

```python
def partition_nodes(grid_size: int, K: int, method: str = "grid") -> list[np.ndarray]:
    """Return K arrays of global node indices, one per drone."""

def build_local_subgraph(data: Data, node_indices: np.ndarray) -> Data:
    """Return a PyG Data with re-indexed nodes; only edges where both
    endpoints are in node_indices are kept. Preserves x and y."""
```

### Acceptance criteria
- [x] `partition_nodes` returns exactly K non-empty arrays
- [x] Union of all arrays equals `range(grid_size**2)` (no node is missing or duplicated)
- [x] `build_local_subgraph` edge_index contains only edges internal to the partition
- [x] Re-indexed node IDs run from 0 to len(node_indices)-1
- [x] Works for K in {2, 3, 4, 5} and grid_size in {10, 50, 100}
- [x] `__main__` block prints partition sizes and a sample subgraph summary
- [x] `train.py` private `_build_local_subgraph` is removed and replaced with an import from `partition.py`
- [x] `federated_agent.py` inline subgraph construction is consolidated to use `build_local_subgraph` from `partition.py`
